pkg/pipeline/BuildPipelineConfigService.go

```diff
@@ -1792,10 +1793,17 @@ func (impl *PipelineBuilderImpl) RetrieveArtifactsByCDPipeline(pipeline *pipelin
 		parentCdId = parentId
 	}
 
-	if stage == bean2.CD_WORKFLOW_TYPE_DEPLOY && len(pipeline.PreStageConfig) > 0 {
-		// Parent type will be PRE for DEPLOY stage
-		parentId = pipeline.Id
-		parentType = bean2.CD_WORKFLOW_TYPE_PRE
+	if stage == bean2.CD_WORKFLOW_TYPE_DEPLOY {
+		pipelineStage, err := impl.pipelineStageRepository.GetCdStageByCdPipelineIdAndStageType(pipeline.Id, repository3.PIPELINE_STAGE_TYPE_PRE_CD)
```

Comment:
do the same thing if parent is POST_CD, needs to be updated in RetrieveParentDetails